chainercv/evaluations/eval_instance_segmentation_voc.py

```diff
@@ -0,0 +1,200 @@
+            an object mask whose shape is :math:`(R, H, W)`. Note that the
+            number of masks :math:`R` in each image does not need to be
+            same as the number of corresponding predicted masks.
+        gt_labels (iterable of numpy.ndarray): An iterable of ground truth
+            labels which are organized similarly to :obj:`gt_masks`. Its
+            length is :math:`N`.
+        iou_thresh (float): A prediction is correct if its Intersection over
+            Union with the ground truth is above this value.
+
+    Returns:
+        tuple of two lists:
+        This function returns two lists: :obj:`prec` and :obj:`rec`.
+
+        * :obj:`prec`: A list of arrays. :obj:`prec[l]` is precision \
+            for class :math:`l`. If class :math:`l` does not exist in \
+            either :obj:`pred_labels` or :obj:`gt_labels`, :obj:`prec[l]` is \
+            set to :obj:`None`.
+        * :obj:`rec`: A list of arrays. :obj:`rec[l]` is recall \
+            for class :math:`l`. If class :math:`l` that is not marked as \
+            difficult does not exist in \
+            :obj:`gt_labels`, :obj:`rec[l]` is \
+            set to :obj:`None`.
+
+    """
+
+    pred_labels = iter(pred_labels)
+    pred_scores = iter(pred_scores)
+    gt_labels = iter(gt_labels)
+
+    n_pos = defaultdict(int)
+    score = defaultdict(list)
+    match = defaultdict(list)
+
+    for pred_mask, pred_label, pred_score, gt_mask, gt_label in \
+            six.moves.zip(
+                pred_masks, pred_labels, pred_scores,
+                gt_masks, gt_labels):
+
+        for l in np.unique(np.concatenate((pred_label, gt_label)).astype(int)):
+            pred_keep_l = pred_label == l
+            pred_mask_l = pred_mask[pred_keep_l]
+            pred_score_l = pred_score[pred_keep_l]
+            # sort by score
+            order = pred_score_l.argsort()[::-1]
+            pred_mask_l = pred_mask_l[order]
+            pred_score_l = pred_score_l[order]
+
+            gt_keep_l = gt_label == l
+            gt_mask_l = gt_mask[gt_keep_l]
+
+            n_pos[l] += gt_keep_l.sum()
+            score[l].extend(pred_score_l)
+
+            if len(pred_mask_l) == 0:
+                continue
+            if len(gt_mask_l) == 0:
+                match[l].extend((0,) * pred_mask_l.shape[0])
+                continue
+
+            iou = mask_iou(pred_mask_l, gt_mask_l)
+            gt_index = iou.argmax(axis=1)
+            # set -1 if there is no matching ground truth
+            gt_index[iou.max(axis=1) < iou_thresh] = -1
+            del iou
+
+            selec = np.zeros(gt_mask_l.shape[0], dtype=bool)
+            for gt_idx in gt_index:
+                if gt_idx >= 0:
+                    if not selec[gt_idx]:
+                        match[l].append(1)
+                    else:
+                        match[l].append(0)
+                    selec[gt_idx] = True
+                else:
+                    match[l].append(0)
+
```

Comment:
I think it is same to insert error check like `eval_detection_voc`
https://github.com/chainer/chainercv/blob/master/chainercv/evaluations/eval_detection_voc.py#L214

Reply:
I updated